docs/docs/vm-fundamentals/hints.md

```diff
@@ -2,6 +2,236 @@
 sidebar_position: 3
 ---
 
-# Hints
+# Hint
+  ## HINTS  IN CAIRO VIRTUAL MACHINE
+Hints in Cairo are pieces of Python code only seen and executed by the prover and are not included in the bytecode. They instruct the prover on how to handle nondeterministic instructions. These nondeterministic instructions are programs that have different outcomes at execution.   
+Hints are guides that developers can provide to the Cairo virtual machine to optimize execution or provide additional information about the program. These hints help improve performance or help the VM make better decisions during the execution of programs. 
+
+## TYPES OF HINTS IN CARIO VIRTUAL MACHINE
+The hints in Cairo Virtual Machine that influence the behavior of programs include:
+### EXECUTION HINTS
+ they guide the execution of cairo programs. They include guides on loop unrolling, parallel execution, or optimizations to enhance. It influences how the VM schedules and executes instructions, thereby improving performance.
+In Cairo, execution hints use the *hint* keyword, followed by a *hint name* and *a value*.
+
+ @example:
+
+ ```cairo  
+      func main() -> felt*:
+      // some code here
+      hint memory_behavior = MemoryBehavior.Sequential;
+      // more code here
+  ```
+  The memory_behavior hint informs the Cairo compiler that the code following the hint   
+ sequentially accesses memory.
+
```

Comment:
Are you sure?  I think hints are enclosed in a block defined by the following symbols `%{` and `}%`